dvalin-tools/dvalin_tools/models/network.py

```diff
@@ -40,6 +39,22 @@ class LinkType(Enum):
     MALFORMED = auto()
 
 
+KNOWN_MALFORMED_URLS: Final[dict[str, tuple[str, LinkType]]] = {
+    "http://「HoYo Quiz」参加方法及びルール公開！": (
+        "https://www.hoyolab.com/article/5773278",
+        LinkType.HOYOLAB,
+    ),
+    "http://xn--hoyo%20quiz%21-bg3kza44m42ca91ar455c49e8xgcfr733b9w9aqyzr/": (
+        "https://www.hoyolab.com/article/5773278",
+        LinkType.HOYOLAB,
+    ),
+    ">https://www.youtube.com/c/GenshinImpact": (
+        "https://www.youtube.com/c/GenshinImpact",
+        LinkType.YOUTUBE,
+    ),
+}
```

Comment:
Maybe a pattern match for this should be prefered over hardcoding each url

Reply:
That's the only known malformed URLs. I wanted to only target the specific ones. Also, it makes us keep a record of the ones that were malformed, and why they were malformed.
Therefore, no pattern matching that could have modified other ones.